a3p-integration/proposals/z:acceptance/test-lib/sync-tools.js

```diff
@@ -0,0 +1,285 @@
+  });
+
+/**
+ * From https://github.com/Agoric/agoric-sdk/blob/442f07c8f0af03281b52b90e90c27131eef6f331/multichain-testing/tools/sleep.ts#L24
+ *
+ * @param {() => Promise} operation
+ * @param {(result: any) => boolean} condition
+ * @param {string} message
+ * @param {RetyrOptions} options
+ * @returns
+ */
+const retryUntilCondition = async (
+  operation,
+  condition,
+  message,
+  { maxRetries = 6, retryIntervalMs = 3500, log, setTimeout },
+) => {
+  console.log({ maxRetries, retryIntervalMs, message });
+  let retries = 0;
+
+  while (retries < maxRetries) {
+    try {
+      const result = await operation();
+      log('RESULT', result);
+      if (condition(result)) {
+        return result;
+      }
+    } catch (error) {
+      if (error instanceof Error) {
+        log(`Error: ${error.message}`);
+      } else {
+        log(`Unknown error: ${String(error)}`);
+      }
+    }
+
+    retries++;
+    console.log(
+      `Retry ${retries}/${maxRetries} - Waiting for ${retryIntervalMs}ms for ${message}...`,
+    );
+    await sleep(retryIntervalMs, { log, setTimeout });
+  }
+
+  throw Error(`${message} condition failed after ${maxRetries} retries.`);
+};
+
+/**
+ * Making sure a core-eval resulted succesfully deploying a contract
+ */
+const makeGetInstances = follow => async () => {
+  const instanceEntries = await follow(
+    '-lF',
+    `:published.agoricNames.instance`,
+  );
+
+  return Object.fromEntries(instanceEntries);
+};
+
+/**
+ *
+ * @param {string} contractName
+ * @param {{follow: () => object, setTimeout: (object) => void}} ambientAuthroity
+ * @param {RetyrOptions} options
+ * @returns
+ */
+export const waitUntilContractDeplyed = (
+  contractName,
+  ambientAuthroity,
+  options,
+) => {
+  const { follow, setTimeout } = ambientAuthroity;
+  const getInstances = makeGetInstances(follow);
+  const {
+    maxRetries = 6,
+    retryIntervalMs = 3500,
+    log = console.log,
+    errorMessage = 'Error',
+  } = options;
+
+  return retryUntilCondition(
+    getInstances,
+    instanceObject => Object.keys(instanceObject).includes(contractName),
+    errorMessage,
+    // @ts-ignore
+    { maxRetries, retryIntervalMs, log, setTimeout },
+  );
+};
+
+// Making sure an account is successfully funded with vbank assets like IST, BLD etc.
+
+const makeQueryCosmosBalace = queryCb => async dest => {
+  const conins = await queryCb('bank', 'balances', dest);
+  return conins.balances;
+};
+
+/**
+ *
+ * @param {Array} balances
+ * @param {CosmosBalanceThresold} thresold
+ * @returns {boolean}
+ */
+const checkCosmosBalance = (balances, thresold) => {
+  const balance = [...balances].find(({ denom }) => denom === thresold.denom);
+  return Number(balance.amount) >= thresold.value;
+};
+
+/**
+ * @param {string} destAcct
+ * @param {{query: () => Promise<object>, setTimeout: (object) => void}} ambientAuthroity
+ * @param {{denom: string, value: number}} threshold
+ * @param {RetyrOptions} options
+ * @returns
```

Comment:
add a type or drop the line.  (I don't care which.)

Reply:
resolved in https://github.com/Agoric/agoric-sdk/pull/10171/commits/d3f245300f2d42b83bc761c1e22950feeacd63db